### DISCUSSION TITLE
Notify user If they want enable screenshot on firefox

### DISCUSSION THREAD

**Discussion by barraIhsan** [monkeytypegame/monkeytype#4405] [Ideas] [answered]
So if you are using Firefox, you'll know that if you press the screenshot button, it will open the images in new tab and causes this "notification"

> Could not save image to clipboard. Opening in new tab instead (make sure popups are allowed)

This is because `ClipboardItem` is not enabled by default in firefox

<details><summary>Click to show image</summary>

![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/7609f9f7-3b37-4d21-b98f-495f24f1a598)
</details>

We can enable it using pref (`about:config`) `dom.events.asyncClipboard.clipboardItem`.

Now, is there a way to somehow tell the user that this feature is disabled in firefox by default? Maybe change the "notification" text?

## Note
This can changed in the future, see [the bug](https://bugzil.la/1619947). This options can also be a site permission in the future, as [this bug suggested](https://bugzil.la/1809685). So maybe in the future (if its implemented), you have to request for this permission first.

**Comment by Miodec** [comment]
Sure thing, I'll add another notification.

**Reply by barraIhsan** [reply]
![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/c1849809-20c2-4184-88b3-adc4cbc72889)
Unrelated issue/discussion, but I dont think this worth its own issue/discussion. Is this a typo? `imgage`

**Reply by Miodec** [reply]
Yeah it is

**Reply by Miodec** [reply, marked as answer]
How about this? Extra message stays up for 10 seconds and only shows up once per session

<img width="645" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/b4a32dca-3687-4795-ad1b-2cd79dbe83fe">

**Reply by barraIhsan** [reply]
That's good? Maybe it's too long for a notification?
People probably can't see it because it switched to the blob image tab when you click the button.
Also, it's not the full pref. name, maybe putting it somewhere?

I can't think other way other than modifying the tooltip (if Firefox detected using UA)
![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/c71f0e00-daa8-4f4b-9f70-69599b9838c1)

or adding FAQ? and then change the notification to be

> This feature is disabled by default in firefox, please see to the [FAQ](link) to enable the feature

**Reply by Miodec** [reply]
I cut down the pref name to save on the size of the notification and improve readability (if you search for `asyncClipboard.clipboardItem` you get one result). The notification has to be up longer because as you said people will be in the other tab looking at the screenshot and need time to read it fully (i was thinking of increasing it even more or even making the notification sticky, but im not sure)

**Reply by barraIhsan** [reply]
hmm.. Or you should add the banner (like the merchandise banner) for firefox user, about this pref

![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/f43d7a09-e409-42de-9717-ed56115d2a2d)

Maybe add it like this (dev monkeytype), so people actually read it?

or maybe notification like this at start (if firefox detected)
![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/df08f1ba-b101-44b8-b470-e1376d91e18e)

**Reply by barraIhsan** [reply]
probably the banner would be better imo

**Reply by Miodec** [reply]
I think the banner is a little bit too intrusive

**Reply by barraIhsan** [reply]
> I think the banner is a little bit too intrusive

Yeah, or what about only display the banner just if the person clicks the screenshot button and it failed?

<details><summary>(in this if statement)</summary>

![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/d46a084a-0a01-4eee-aee8-ebb0e7ef186b)

</details>

**Reply by Miodec** [reply]
> Yeah, or what about only display the banner just if the person clicks the screenshot button and it failed?

Thats how it works now, the message only shows if copying straight to the clipboard failed.

**Reply by barraIhsan** [reply]
Ok, but it does not actually show the notification for me? (new profile firefox)
![image](https://github.com/monkeytypegame/monkeytype/assets/57800056/0d749827-f12c-4241-bb1a-d6cd3f635675)
~~(the first if statement should result true, because its true for me (`navigator.userAgent.toLowerCase().indexOf("firefox") > -1`), but not sure the second one)~~

**Reply by Miodec** [reply]
The change is not deployed yet

**Reply by barraIhsan** [reply]
oh ok, sorry.
So you gonna end up with the notifications?

**Comment by barraIhsan** [comment]
Update: [Firefox 127.0](https://www.mozilla.org/en-US/firefox/127.0/releasenotes/) and above now supports clipboard read & write by default ([see supported browser on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API#browser_compatibility))
Which means, `dom.events.asyncClipboard.clipboardItem` no longer defaults to `false`
